File: Trees/Easy/110/110.c
#include <stdio.h>
#include <stdlib.h>

// Definition for a binary tree node.
struct TreeNode {
    int val;
    struct TreeNode *left;
    struct TreeNode *right;
};

int dfs(struct TreeNode *root){
    if(root == NULL){
        return 0;
    }
    int leftHeight = dfs(root->left);
    int rightHeight = dfs(root->right);
    if(abs(leftHeight - rightHeight) > 1 || leftHeight == -1 || rightHeight == -1){
        return -1;
    }
    int max = leftHeight > rightHeight ? leftHeight : rightHeight;
    return 1 + max;
}

int isBalanced(struct TreeNode* root){
    return dfs(root) != -1;
}

int main(){}

File: Trees/Easy/110.BalancesBinaryTree/110.py
from typing import Optional

# Definition for a binary tree node.
class TreeNode:
    def __init__(self, val=0, left=None, right=None):
        self.val = val
        self.left = left
        self.right = right

class Solution:
    def isBalanced(self, root: Optional[TreeNode]) -> bool:
        def dfs(root):
            if not root:
                return 0
            
            leftHeight = dfs(root.left)
            rightHeight = dfs(root.right)
            if abs(leftHeight - rightHeight) > 1 or leftHeight == -1 or rightHeight == -1:
                return -1
            return 1 + max(leftHeight, rightHeight)
        return dfs(root) != -1